ultralytics/models/yolo/classify/train.py

```diff
@@ -196,16 +196,18 @@ def label_loss_items(self, loss_items: torch.Tensor | None = None, prefix: str =
 
     def final_eval(self):
         """Evaluate trained model and save validation results."""
-        for f in self.last, self.best:
-            if f.exists():
-                strip_optimizer(f)  # strip optimizers
-                if f is self.best:
-                    LOGGER.info(f"\nValidating {f}...")
-                    self.validator.args.data = self.args.data
-                    self.validator.args.plots = self.args.plots
-                    self.metrics = self.validator(model=f)
-                    self.metrics.pop("fitness", None)
-                    self.run_callbacks("on_fit_epoch_end")
+        with torch_distributed_zero_first(LOCAL_RANK):  # strip only on GPU 0; other GPUs should wait
+            if RANK in {-1, 0}:
+                for f in self.last, self.best:
+                    if f.exists():
+                        strip_optimizer(f)  # strip optimizers
+
+        LOGGER.info(f"\nValidating {self.best}...")
```

Comment:
@Y-T-G this might error is self.best doesn't exist. Current code checks that it exists first.